### PR TITLE
[Feature] Support HorNet Backbone for dev1.x

### PR DESCRIPTION
**Pull request by okotaku**
## Motivation

Related PR: https://github.com/open-mmlab/mmclassification/pull/1013

## Checklist

**Before PR**:

- [x] Pre-commit or other linting tools are used to fix the potential lint issues.
- [x] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by mzr1996** [comment]
Please merge the latest `dev-1.x` branch.
Because #1064 is merged, please remove the `num_classes` argument in the `Mixup` and `CutMix`, and the argument is handled by `data_preprocessor` now.

File: configs/hornet/hornet-base-gf_8xb64_in1k.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    '../_base_/models/hornet/hornet-base-gf.py',
+    '../_base_/datasets/imagenet_bs64_swin_224.py',
+    '../_base_/schedules/imagenet_bs1024_adamw_swin.py',
+    '../_base_/default_runtime.py',
+]
+
+data = dict(samples_per_gpu=64)
+
+optimizer = dict(lr=4e-3)
+optimizer_config = dict(grad_clip=dict(max_norm=1.0), _delete_=True)
```
Comment (Ezra-Yu) [inline configs/hornet/hornet-base-gf_8xb64_in1k.py:11]:
```suggestion
optim_wrapper=dict(
     optimizer = dict(lr=4e-3),
     clip_grad=dict(max_norm=100.0)
)
```

File: configs/hornet/hornet-base_8xb64_in1k.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    '../_base_/models/hornet/hornet-base.py',
+    '../_base_/datasets/imagenet_bs64_swin_224.py',
+    '../_base_/schedules/imagenet_bs1024_adamw_swin.py',
+    '../_base_/default_runtime.py',
+]
+
+data = dict(samples_per_gpu=64)
+
+optimizer = dict(lr=4e-3)
+optimizer_config = dict(grad_clip=dict(max_norm=5.0), _delete_=True)
```
Comment (Ezra-Yu) [inline configs/hornet/hornet-base_8xb64_in1k.py:11]:
```suggestion
optim_wrapper=dict(
     optimizer = dict(lr=4e-3),
     clip_grad=dict(max_norm=100.0)
)
```

File: configs/hornet/hornet-small-gf_8xb64_in1k.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    '../_base_/models/hornet/hornet-small-gf.py',
+    '../_base_/datasets/imagenet_bs64_swin_224.py',
+    '../_base_/schedules/imagenet_bs1024_adamw_swin.py',
+    '../_base_/default_runtime.py',
+]
+
+data = dict(samples_per_gpu=64)
+
+optimizer = dict(lr=4e-3)
+optimizer_config = dict(grad_clip=dict(max_norm=1.0), _delete_=True)
```
Comment (Ezra-Yu) [inline configs/hornet/hornet-small-gf_8xb64_in1k.py:11]:
```suggestion
optim_wrapper=dict(
     optimizer = dict(lr=4e-3),
     clip_grad=dict(max_norm=100.0)
)
```

**Review by Ezra-Yu — CHANGES_REQUESTED** [review]
The old-style optimizer has been deprecated, Now use the [optim_wrapper](https://mmclassification.readthedocs.io/en/1.x/advanced_guides/schedule.html).

**Comment by okotaku** [comment]
@Ezra-Yu Thank you!

**Review by Ezra-Yu — APPROVED** [review]
LGTM.

**Pull request merged**